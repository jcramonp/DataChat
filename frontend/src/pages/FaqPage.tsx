import { useTranslation } from "react-i18next";
import "./MainPage.css"; // reutilizamos estilos (container, etc.)

type FaqItem = { q: string; a: string };

export default function FaqPage() {
  const { t } = useTranslation();
  const items = t("faq.items", { returnObjects: true }) as FaqItem[];

  return (
    <main className="container mt-16">
      <section className="chat-card" style={{ borderRadius: 16 }}>
        <h1 style={{ margin: 0 }}>{t("faq.title")}</h1>
        <p style={{ color: "#6b7280" }}>{t("faq.intro")}</p>

        <div style={{ display: "grid", gap: 10, marginTop: 16 }}>
          {items.map((it, i) => (
            <details key={i} style={{ background: "#fff", borderRadius: 12, padding: "10px 14px" }}>
              <summary style={{ cursor: "pointer", fontWeight: 600 }}>{it.q}</summary>
              <p style={{ marginTop: 8, color: "#374151" }}>{it.a}</p>
            </details>
          ))}
        </div>
      </section>
    </main>
  );
}
