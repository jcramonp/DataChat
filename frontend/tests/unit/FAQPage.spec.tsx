import { renderWithI18n } from "../utils/renderWithI18n";
import FaqPage from "@/pages/FaqPage";

it("estructura estable (snapshot)", () => {
  const { container } = renderWithI18n(<FaqPage />, "es");
  expect(container).toMatchSnapshot();
});
