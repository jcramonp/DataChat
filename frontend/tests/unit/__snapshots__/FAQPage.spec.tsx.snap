// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`estructura estable (snapshot) 1`] = `
<div>
  <main
    class="container mt-16"
  >
    <section
      class="chat-card"
      style="border-radius: 16px;"
    >
      <h1
        style="margin: 0px;"
      >
        Preguntas frecuentes
      </h1>
      <p
        style="color: rgb(107, 114, 128);"
      >
        faq.intro
      </p>
      <div
        style="display: grid; gap: 10px; margin-top: 16px;"
      >
        <details
          style="background: rgb(255, 255, 255); border-radius: 12px; padding: 10px 14px;"
        >
          <summary
            style="cursor: pointer; font-weight: 600;"
          >
            ¿Cómo conecto mi base?
          </summary>
          <p
            style="margin-top: 8px; color: rgb(55, 65, 81);"
          >
            Ve a Admin → Conexiones.
          </p>
        </details>
        <details
          style="background: rgb(255, 255, 255); border-radius: 12px; padding: 10px 14px;"
        >
          <summary
            style="cursor: pointer; font-weight: 600;"
          >
            ¿Excel soportado?
          </summary>
          <p
            style="margin-top: 8px; color: rgb(55, 65, 81);"
          >
            Sí, xlsx/csv para demo.
          </p>
        </details>
      </div>
    </section>
  </main>
</div>
`;
